### PR TITLE
Fix 3985 add catalog autosearch

### PR DESCRIPTION
**Pull request by MV88**
## Description
This pr adds AutoSearch to the filter of catalog form
Also loading state has been moved from local to the global state
The delay can be configured in the initialState of localConfig, for example: 
```
"initialState": {
  "defaultState": {
    "catalog": {
      "delayAutoSearch" : 1000
    }
  }
}
```

## Issues
 - #3985

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
When you type in the filter you have to click on "search button" in order to trigger the search

**What is the new behavior?**
after 1 second it triggers automatically the search when idle for 1s

**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No 

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
notes this could be harmful for projects that are customizing the catalog plugin (components and plugin changes needs to be ported too)

### CODE REVIEW

File: web/client/epics/catalog.js [hunk 1]
```diff
@@ -162,5 +172,19 @@ module.exports = (API) => ({
...
+                const state = getState();
+                const delay = delayAutoSearchSelector(state);
+                return Rx.Observable.timer(delay);
+            })
+            .switchMap(({text}) => {
+                const state = getState();
+                const pageSize = pageSizeSelector(state);
+                const {type, url} = selectedCatalogSelector(state);
+                return Rx.Observable.of(textSearch(type, url, 1, pageSize, text))
+                    .startWith(setLoading(true));
```
Comment (mbarto) [inline web/client/epics/catalog.js]: Is this needed? It looks like textSearch already dispatches setLoading(true)

File: web/client/reducers/catalog.js [hunk 2]
```diff
@@ -59,15 +60,17 @@ function catalog(state = {
 }, action) {
```
Comment (mbarto) [inline web/client/reducers/catalog.js]: I would add the new state property loading here too

Comment (mbarto) [reply web/client/reducers/catalog.js]: And also pageSize, maybe?

File: web/client/epics/catalog.js (hunk 1 above)
Comment (MV88) [reply web/client/epics/catalog.js]: yes, on a second thought this can be removed

File: web/client/reducers/catalog.js [hunk 3]
```diff
@@ -54,6 +54,8 @@ function catalog(state = {
         selectedService: "",
         newService: {}
     },
+    delayAutoSearch: 1000,
```
Comment (mbarto) [inline web/client/reducers/catalog.js]: and loading?

File: web/client/epics/__tests__/catalog-test.js [hunk 4]
```diff
@@ -36,38 +36,5 @@ describe('catalog Epics', () => {
         });
 
     });
-    it('autoSearchEpic', (done) => {
```
Comment (mbarto) [inline web/client/epics/__tests__/catalog-test.js]: Why this has been removed?

Comment (MV88) [reply web/client/epics/__tests__/catalog-test.js]: it was dispatching only a thunk that is tested in the actions/__tests__/catalog-test.js
one way to test it here is to transform the thunk into an epic, if you think is worth the shot, i'll do it

**Pull request merged**